Cognigy/Webchat
# Fix issues with controlCommands messages: QR disabling and messages collation

# Success criteria
- Fix a bug where "setRating" controlCommand type was not filter out from messages list and hasReply parameter. For this reason after rating submit, if last message is a QR, all buttons get disabled.
- Fix a bug where lastMessage logic was not excluding the controlCommads messages. For this reason the messages collation was not working in case of controlCommands are sent in between of two bot messages.

Both bugs are visible in this screenshot, where it was submitted the privacy message and also the rating:

![Screenshot from 2024-08-21 13-18-39](https://github.com/user-attachments/assets/edd20a60-d2a1-4f3e-a493-ca8900ad5ccd)

# How to test
1. Use this endpoint: https://endpoint-dev.cognigy.ai/51083847963e64f271308e3e940057f8a353b80e4a5579d96dec292202a6c236 and check success criteria
